knimin/lib/data_access.py

```diff
@@ -3191,6 +3186,252 @@ def delete_sample_plate(self, id):
             TRN.add(sql, [id])
             TRN.execute()
 
+    def get_id_by_name(self, field, name):
```

Comment:
@qiyunzhu understood. However this just points out how important is to break the PRs in single functionality. If I remember correctly, when we discussed the path to follow for the implementation of this I specifically asked you to add only the page for creating the plate map, and in another PR add the listing page. That would've reduced the amount of code to review at once. It is ok now that this PR is done but please keep in mind for future PR.